src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Collect.cs

```diff
@@ -0,0 +1,464 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.DLP;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.IR.Symbols;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using RecordNode = Microsoft.PowerFx.Core.IR.Nodes.RecordNode;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // Collect(collection:*[...], item1:![...]|*[...], ...)
+    internal class CollectFunction : BuiltinFunction
+    {
+        public override bool AffectsCollectionSchemas => true;
+
+        public bool CanSuggestThisItem => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public override bool ManipulatesCollections => true;
+
+        public override bool ModifiesValues => true;
+
+        public override bool IsSelfContained => false;
+
+        public override bool RequiresDataSourceScope => true;
+
+        public virtual bool IsScalar => false;
+
+        public override bool CanSuggestInputColumns => true;
+
+        public override bool MutatesArg0 => true;
+
+        public override RequiredDataSourcePermissions FunctionPermission => RequiredDataSourcePermissions.Create;
+
+        /// <summary>
+        /// Since Arg1 and Arg2 depends on type of Arg1 return false for them.
+        /// </summary>
+        public override bool TryGetTypeForArgSuggestionAt(int argIndex, out DType type)
+        {
+            if (argIndex == 0)
+            {
+                type = DType.EmptyTable;
+                return true;
+            }
```

Comment:
This differs from PA. PA handles suggestions for functions with `ParmTypes == 0` differently.